src/index.rs

```diff
@@ -403,7 +403,15 @@ impl ResourceIndex {
         log::debug!("Updating a single entry in the index");
 
         if !path.as_ref().exists() {
-            return self.forget_id(old_id);
+            if let Some(indexed_path) = self.id2path.get(&old_id) {
+                if indexed_path.as_path() == path.as_ref() {
+                    return self.forget_id(old_id);
+                }
+            }
+
+            return Err(ArklibError::Path(
+                "The path isn't indexed or doesn't map to the id".into(),
+            ));
```

Comment:
We throw this error if the caller provided wrong `old_id`. Alternative would be querying `path2id` by `path`, but this would not cover the case with resources that were indexed but then disappeared from the filesystem.

Alternatively, we can extract such case into `unindex_one`. The caller would need to be more precise about what to call when, but it's always possible. The app that calls the `update_one` or `index_new` function must always have information, otherwise complete re-indexing with `update_all` must be performed.

Usage scenarios:
1. User scrolls through images in Gallery-like app, presses `delete` button. We don't want to run full re-indexing because it's slow. With reactive updates the index must be up-to-date anyway. We precisely know that only this single resource requires update and we call `update_one` (or `unindex_one`).
2. User scrolls through images in Gallery again, now they press `edit` button. We pass the image to another app but we have callback when we come back to Gallery. In such a fallback, `update_one` must be called. The app always has old id and path.
3. User opens graphical editor, creates an image from scratch, selects ARK-enabled folder, so we have an index in there by path `.ark/index`. Function `index_new` can be called to add single path to the index. However, I'm not entirely sure this use-case is useful. Because, other apps either monitor the folder at the same moment and catch update automatically, or they don't monitor and will perform full re-indexing anyway. So `index_new` might be useless.

Reply:
3'. User uses Navigator to copy a batch of files from one root to another root. Navigator can use `unindex_one` and `index_new` to update the index in both source and destination folders. However, again, with reactive updates this should be done automatically. Is it worth adding new functions?